uber/templates/macros.html

```diff
@@ -0,0 +1,115 @@
+    inner_text = inner_text.replace("'", "&quot;")
+    return "<a onClick='{inner_text}' href='{self.href}'>{self.text}</a>".format(inner_text=inner_text, self=self)
+    #}
+{%- endmacro %}
+
+
+{% macro checkbox(model, field_name, id=None) -%}
+    <input
+        type="checkbox"
+        name="{{ field_name }}"
+        value="1"
+        {% if id %}id="{{ id }}"{% endif %}
+        {% if model[field_name] %}checked{% endif %} />
+{%- endmacro %}
+
+
+{% macro checkgroup(model, field_name, no_instance=False) -%}
+    {% set defaults = model[field_name] if not no_instance else None %}
+    {% set defaults = defaults.split(",") if defaults else [] %}
+
+    {% for num, desc in model.get_field(field_name).type.choices -%}
+        <label style="font-weight: normal;">
+            <input
+                type="checkbox"
+                name="{{ field_name }}"
+                value="{{ num }}"
+                {% if num|string in defaults %}checked{% endif %} />
+           {{ desc }}
+        </label>
+        {% if not loop.last %}&nbsp;&nbsp;{% endif %}
+    {%- endfor %}
+{%- endmacro %}
+
+
+{% macro radiogroup(model, field_name, onchanged="donationChanged();") -%}
+    {# TODO: remove hardcoded "donationChanged()" param #}
+
+    {% set default = model[field_name] %}
+
+    {% for num, desc in model.get_field(field_name).type.choices -%}
+        <label class="btn btn-default" style="text-align: left;">
+            <input
+                type="radio"
+                autocomplete="off"
+                name="{{ field_name }}"
+                value="{{ num }}"
+                onchange="{{ onchanged }}"
+                {% if num|string == defaults %}checked{% endif %} />
+            {{ desc }}
+        </label>
+    {%- endfor %}
+{%- endmacro %}
+
+
+{% macro nav_menu(model, current_page_path) -%}
+    {% if not model.is_new -%}
+        <table class="menu"><tr>
+        {% for href, label, display in varargs|batch(3, False) if display %}
+            {% set href = href.format(**model.__dict__) %}
+            <td width="{{ 100 // loop.length }}%">
+                {%- if current_page_path.endswith(href.split('?')[0]) -%}
+                    {{ label }}
+                {%- else -%}
+                    <a href="{{ href }}">{{ label }}</a>
+                {%- endif -%}
+            </td>
+        {% endfor %}
+        </tr></table>
+    {%- endif %}
+{%- endmacro %}
+
+
+{% macro checklist_image(enabled) -%}
+    <img
+        src="../static/images/checkbox_{% if not enabled %}un{% endif %}checked.png"
+        style="vertical-align:top ; margin-right:5px"
+        height="20"
+        width="20" />
+{%- endmacro %}
+
+
+{% macro radio(name, value, checked) -%}
+    {# TODO: BROKEN: make this look more like the checkbox macro, prob don't need to pass in all the params. #}
+    <div class="radio">
```

Comment:
need to remove the TODO by verifying that either this is no longer broken, or fix what the problem was